src/hotspot/share/runtime/jniHandles.hpp

```diff
@@ -59,18 +62,24 @@ class JNIHandles : AllStatic {
   static bool current_thread_in_native();
 
  public:
-  // Low tag bit in jobject used to distinguish a jweak.  jweak is
-  // type equivalent to jobject, but there are places where we need to
-  // be able to distinguish jweak values from other jobjects, and
-  // is_weak_global_handle is unsuitable for performance reasons.  To
-  // provide such a test we add weak_tag_value to the (aligned) byte
-  // address designated by the jobject to produce the corresponding
-  // jweak.  Accessing the value of a jobject must account for it
-  // being a possibly offset jweak.
-  static const uintptr_t weak_tag_size = 1;
-  static const uintptr_t weak_tag_alignment = (1u << weak_tag_size);
-  static const uintptr_t weak_tag_mask = weak_tag_alignment - 1;
-  static const int weak_tag_value = 1;
+  // Low tag bits in jobject used to distinguish its type. Checking
+  // the underlying storage type is unsuitable for performance reasons.
+  enum TypeTag {
+    local = 0b00,
+    weak = 0b01,
```

Comment:
Maybe `jweak`? Just to be less ambiguous.

Reply:
I think I should fix all the references, currently we use `global weak`, `weak global`, `weak` and `jweak`. jweak belongs to C, it is just a typedef to make the API signatures more clear (it is the same type as jobject). And use `weak global` everywhere which is how the C API actually names the types:
```c
typedef enum _jobjectType {
     JNIInvalidRefType    = 0,
     JNILocalRefType      = 1,
     JNIGlobalRefType     = 2,
     JNIWeakGlobalRefType = 3
} jobjectRefType;
```